language-plutus-core/src/Language/PlutusCore/Evaluation/Machine/ExBudgeting.hs

```diff
@@ -146,17 +161,83 @@ instance PrettyBy config ExBudgetState where
 newtype ExTally = ExTally (MonoidalHashMap ExBudgetCategory ExBudget)
     deriving stock (Eq, Generic, Show)
     deriving (Semigroup, Monoid) via (GenericSemigroupMonoid ExTally)
+    deriving anyclass NFData
 instance PrettyBy config ExTally where
     prettyBy config (ExTally m) =
         parens $ fold (["{ "] <> (intersperse (line <> "| ") $ fmap group $
           ifoldMap (\k v -> [(prettyBy config k <+> "causes" <+> prettyBy config v)]) m) <> ["}"])
 
-$(mtraverse makeLenses [''ExBudgetState, ''ExBudget])
-
 -- TODO See language-plutus-core/docs/Constant application.md for how to properly implement this
 estimateStaticStagedCost
     :: BuiltinName -> [WithMemory Value uni] -> (ExCPU, ExMemory)
 estimateStaticStagedCost _ _ = (1, 1)
 
+data CostingFunParameters =
```

Comment:
I wonder whether we should call this `CostModel`? I *think* that's the term we've generally been using for the thing that tells us how to convert our actual operations into abstract costs, so we should probably use the word just to avoid confusion.

Reply:
Aye, will rename.